contracts/forwarder/Eip712Forwarder.sol

```diff
@@ -36,22 +34,28 @@ contract Eip712Forwarder is IForwarder {
         _verifySig(req, domainSeparator, requestTypeHash, suffixData, sig);
     }
 
-    //note that verifyAndCall doesn't re-throw target's call revert, but instead return it as (success,ret).
-    // the nonce is incremented either if the target method reverts or not.
-    function verifyAndCall(ForwardRequest memory req,
-        bytes32 domainSeparator, bytes32 requestTypeHash, bytes memory suffixData, bytes memory sig)
-    public override
-    returns (bool success, bytes memory ret) {
-
+    function execute(
+        ForwardRequest memory req,
+        bytes32 domainSeparator,
+        bytes32 requestTypeHash,
+        bytes memory suffixData,
+        bytes memory sig
+    )
+    public
+    override
+    returns (bool, string memory) {
         _verifyNonce(req);
         _verifySig(req, domainSeparator, requestTypeHash, suffixData, sig);
         _updateNonce(req);
 
         // solhint-disable-next-line avoid-low-level-calls
-        return req.to.call{gas: req.gas}(abi.encodePacked(req.data, req.from));
+        (bool executeSuccess, bytes memory ret) = req.to.call{gas: req.gas}(abi.encodePacked(req.data, req.from));
+        if (!executeSuccess){
+            return (false, GsnUtils.getError(ret));
```

Comment:
the problem with getError is that it adds dependency on LibBytes, which I didn't want to have in the forwarder, and leave it to its caller. why not redefine `ret` as `revertBytes`, and let the caller handle it.